File: webpage/src/core/contracts/property.ts
import { z } from "zod";

export const propertySchema = z.object({
  idProperty: z.string(),
  idOwner: z.string(),
  name: z.string(),
  addressProperty: z.string(),
  priceProperty: z.number(),
  imageUrl: z.string().optional().default(""),
});

export type Property = z.infer<typeof propertySchema>;

export const listResponseSchema = z.object({
  items: z.array(propertySchema),
  total: z.number(),
  page: z.number(),
  pageSize: z.number(),
});

export type PropertyListResponse = z.infer<typeof listResponseSchema>;

File: api/Domain/Entities/Property.cs
namespace Domain.Entities;

public class Property
{
    public string idProperty { get; set; } = "";
    public string idOwner { get; set; } = "";
    public string name { get; set; } = "";
    public string addressProperty { get; set; } = "";
    public decimal priceProperty { get; set; }
    public string codeInternal { get; set; } = "";
    public int year { get; set; }
}
